shell/platform/darwin/macos/framework/Headers/FlutterDartProject.h

```diff
@@ -46,6 +48,52 @@ FLUTTER_DARWIN_EXPORT
  */
 @property(nonatomic, nullable, strong) NSArray<NSString*>* dartEntrypointArguments;
 
+/**
+ * Returns the file name for the given asset. If the bundle with the identifier
+ * "io.flutter.flutter.app" exists, it will try use that bundle; otherwise, it
+ * will use the main bundle.  To specify a different bundle, use
+ * `+lookupKeyForAsset:fromBundle`.
+ *
+ * @param asset The name of the asset. The name can be hierarchical.
+ * @return the file name to be used for lookup in the main bundle.
+ */
++ (NSString*)lookupKeyForAsset:(NSString*)asset;
+
+/**
+ * Returns the file name for the given asset.
+ * The returned file name can be used to access the asset in the supplied bundle.
+ *
+ * @param asset The name of the asset. The name can be hierarchical.
+ * @param bundle The `NSBundle` to use for looking up the asset.
+ * @return the file name to be used for lookup in the main bundle.
+ */
++ (NSString*)lookupKeyForAsset:(NSString*)asset fromBundle:(nullable NSBundle*)bundle;
+
+/**
+ * Returns the file name for the given asset which originates from the specified package.
+ * The returned file name can be used to access the asset in the application's main bundle.
+ *
+ * @param asset The name of the asset. The name can be hierarchical.
+ * @param package The name of the package from which the asset originates.
+ * @return the file name to be used for lookup in the main bundle.
+ */
++ (NSString*)lookupKeyForAsset:(NSString*)asset fromPackage:(NSString*)package;
+
+/**
+ * Returns the file name for the given asset which originates from the specified package.
+ * The returned file name can be used to access the asset in the specified bundle.
+ *
+ * @param asset The name of the asset. The name can be hierarchical.
+ * @param package The name of the package from which the asset originates.
+ * @param bundle The bundle to use when doing the lookup.
+ * @return the file name to be used for lookup in the main bundle.
+ */
++ (NSString*)lookupKeyForAsset:(NSString*)asset
+                   fromPackage:(NSString*)package
+                    fromBundle:(nullable NSBundle*)bundle;
```

Comment:
We're adding enough commonality here that it probably makes sense to switch to a single header with platform #ifdef's instead of two headers, since the goal over time is to merge both headers.

@jmagman and @cbracken do you have any objection to doing that here?

Reply:
I would like to see as many headers shared as possible.  

x-ref note to self: when I tried to swap the shared modules to module imports it also required `#ifdef`s
https://github.com/flutter/flutter/issues/60025#issuecomment-726413256

Reply:
Why is this marked as resolved? There are still two separate headers with a ton of duplication in the current state of the PR, rather than one shared header.

Reply:
Done.